src/tokenizer/tokenizer.py

```diff
@@ -803,10 +818,17 @@ def parse_digits(w):
 def prepare(txt):
     """ Convert txt to Unicode (on Python 2.7) and replace composite glyphs
         with single code points """
-    return UNICODE_REGEX.sub(
+
+    txt = UNICODE_REGEX.sub(
         lambda match: UNICODE_REPLACEMENTS[match.group(0)], make_str(txt)
     )
 
+    # Remove unwanted characters (e.g. soft hyphens, etc.)
+    for char in UNWANTED_CHARACTERS:
+        txt = txt.replace(char, '')
+
```

Comment:
This (the for-loop) could and should be included in the UNICODE_REGEX sub above, so this becomes a single operation.